envoy/http/filter.h

```diff
@@ -1063,6 +1078,35 @@ class FilterChainFactoryCallbacks {
  */
 using FilterFactoryCb = std::function<void(FilterChainFactoryCallbacks& callbacks)>;
 
+/**
+ * Simple struct of additional contextual information of filter factory, e.g. custom filter name
+ * from configuration, canonical filter name, etc.
+ */
+struct FilterFacotryContext {
+  // Custom filter name from configuration.
+  absl::string_view custom_name;
+  // Canonical filter name.
+  absl::string_view filter_name;
```

Comment:
I don't know what a custom filter name and canonical name is. Can you provide some examples?